src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function addedFunc(a) {
+    if (a !== undefined) {
+      sum += a;
+
+      return addedFunc;
+    }
+
+    const newSum = sum;
+
+    sum = 0;
+
+    return newSum;
```

Comment:
Consider adding a comment above the `addedFunc` to explain its behavior. It's a recursive function that either adds to the sum or returns the total sum when called without arguments. This can help others understand the flow of the function more quickly.